NineChronicles.Headless.Executable/Commands/ChainCommand.cs

```diff
@@ -58,5 +59,187 @@ public void Help([FromService] ICoconaHelpMessageBuilder helpMessageBuilder)
+                            hackandslashCount++;
+                        }
+
+                        if (action.InnerAction is HackAndSlash4 hackandslashAction4)
+                        {
+                            hackandslashCount++;
+                        }
+
+                        if (action.InnerAction is HackAndSlash3 hackandslashAction3)
+                        {
+                            hackandslashCount++;
+                        }
+
+                        if (action.InnerAction is HackAndSlash2 hackandslashAction2)
+                        {
+                            hackandslashCount++;
+                        }
+
+                        if (action.InnerAction is HackAndSlash0 hackandslashAction0)
+                        {
+                            hackandslashCount++;
+                        }
+
+                        if (action.InnerAction is MimisbrunnrBattle mimisbrunnrAction)
+                        {
+                            mimisbrunnrCount++;
+                        }
+
+                        if (action.InnerAction is MimisbrunnrBattle4 mimisbrunnrAction4)
+                        {
+                            mimisbrunnrCount++;
+                        }
+
+                        if (action.InnerAction is MimisbrunnrBattle3 mimisbrunnrAction3)
+                        {
+                            mimisbrunnrCount++;
+                        }
+
+                        if (action.InnerAction is MimisbrunnrBattle2 mimisbrunnrAction2)
+                        {
+                            mimisbrunnrCount++;
+                        }
+
+                        if (action.InnerAction is MimisbrunnrBattle0 mimisbrunnrAction0)
+                        {
+                            mimisbrunnrCount++;
+                        }
+
+                        if (action.InnerAction is RankingBattle rankingbattleAction)
+                        {
+                            rankingbattleCount++;
+                        }
+
+                        if (action.InnerAction is RankingBattle5 rankingbattleAction5)
+                        {
+                            rankingbattleCount++;
+                        }
+
+                        if (action.InnerAction is RankingBattle4 rankingbattleAction4)
+                        {
+                            rankingbattleCount++;
+                        }
+
+                        if (action.InnerAction is RankingBattle3 rankingbattleAction3)
+                        {
+                            rankingbattleCount++;
+                        }
+
+                        if (action.InnerAction is RankingBattle2 rankingbattleAction2)
+                        {
+                            rankingbattleCount++;
+                        }
+
+                        if (action.InnerAction is RankingBattle0 rankingbattleAction0)
+                        {
+                            rankingbattleCount++;
+                        }
```

Comment:
```suggestion
                        switch (action.InnerAction)
                        {
                            case HackAndSlash _:
                            case HackAndSlash0 _:
                            case HackAndSlash1 _:
                            case HackAndSlash2 _:
                            case HackAndSlash3 _:
                            case HackAndSlash4 _:
                            case HackAndSlash5 _:
                            case HackAndSlash6 _:
                                hackandslashCount++;
                                break;
                            caee MimisbrunnrBattle  _:
                            caee MimisbrunnrBattle0  _:
                            caee MimisbrunnrBattle2  _:
                            caee MimisbrunnrBattle3  _:
                            caee MimisbrunnrBattle4  _:
                                mimisbrunnrCount++;
                                break;
                            case RankingBattle _:
                            case RankingBattle0 _:
                            case RankingBattle2 _:
                            case RankingBattle3 _:
                            case RankingBattle4 _:
                            case RankingBattle5 _:
                                rankingbattleCount++;
                                break;
                        }
```

Reply:
👍

Reply:
We may need to tag interface for type querying like this... cc @planetarium/9c-dev @planetarium/9c-backend